### PR TITLE
script changes

### PR DESCRIPTION
**Pull request by BlazeMCworld**

### CODE REVIEW

**Review by TechStreetDev — DISMISSED** [review]

**Review by RedVortexDev — DISMISSED** [review]
mongus :D

File: common/src/main/java/io/github/codeutilities/util/ComponentUtil.java [hunk 1]
```diff
@@ -50,4 +51,41 @@ public static TextComponent fromString(String message) {
 
         return result;
     }
+
+    public static String toFormattedString(Component message) {
```
Comment (Reasonlesss) [inline common/src/main/java/io/github/codeutilities/util/ComponentUtil.java:55]: Why?

**Review by Reasonlesss — DISMISSED** [review]

File: common/src/main/java/io/github/codeutilities/util/ComponentUtil.java (hunk 1 above)
Comment (BlazeMCworld) [reply common/src/main/java/io/github/codeutilities/util/ComponentUtil.java:55]: image
because its used here?

**Comment by Reasonlesss** [comment]
This pull request is a bit too volatile imo, not finalized stuff yet

**Review by TechStreetDev — DISMISSED** [review]
Looks good!

File: common/src/main/java/io/github/codeutilities/CodeUtilities.java [hunk 2]
```diff
@@ -19,6 +20,7 @@
     public static final Minecraft MC = Minecraft.getInstance();
     public static final Gson GSON = new Gson();
     public static final Logger LOGGER = LogManager.getLogger();
+    public static final JsonParser JSON_PARSER = new JsonParser();
```
Comment (Reasonlesss) [inline common/src/main/java/io/github/codeutilities/CodeUtilities.java:23]: is this mc built in?

File: common/src/main/java/io/github/codeutilities/menu/ScriptEditorMenu.java [hunk 3]
```diff
@@ -30,26 +31,29 @@ public ScriptEditorMenu(String name) {
...
-                errorMsg.setText("§aScript Valid");
-            } catch (Exception err) {
-                field.textColor = 0xFFFF3333;
-                errorMsg.setText("§c" + err.getMessage());
-            }
-        });
+        panel = new CPanel(150, 110);
+
+        field = new CScriptTextField(script.getSource(), 2, 2, 146, 96, true);
+        errorMsg = new CText( 75, 105, "§aValid", 0, 1, true, true);
```
Comment (Reasonlesss) [inline common/src/main/java/io/github/codeutilities/menu/ScriptEditorMenu.java]: § codes are legacy tmk

**Review by Reasonlesss — CHANGES_REQUESTED** [review]

File: common/src/main/java/io/github/codeutilities/CodeUtilities.java (hunk 2 above)
Comment (BlazeMCworld) [reply common/src/main/java/io/github/codeutilities/CodeUtilities.java:23]: gson

File: common/src/main/java/io/github/codeutilities/menu/ScriptEditorMenu.java (hunk 3 above)
Comment (BlazeMCworld) [reply common/src/main/java/io/github/codeutilities/menu/ScriptEditorMenu.java]: ik but i still like them, if you want you can replace them

File: common/src/main/java/io/github/codeutilities/CodeUtilities.java (hunk 2 above)
Comment (Reasonlesss) [reply common/src/main/java/io/github/codeutilities/CodeUtilities.java:23]: i mean like is this gson provided by mc?

Comment (BlazeMCworld) [reply common/src/main/java/io/github/codeutilities/CodeUtilities.java:23]: i didnt add any dependencies so i think so

Comment (Reasonlesss) [reply common/src/main/java/io/github/codeutilities/CodeUtilities.java:23]: alright just making sure

File: common/src/main/java/io/github/codeutilities/menu/ScriptEditorMenu.java (hunk 3 above)
Comment (Reasonlesss) [reply common/src/main/java/io/github/codeutilities/menu/ScriptEditorMenu.java]: I do not want to use the legacy stuff

Comment (TechStreetDev) [reply common/src/main/java/io/github/codeutilities/menu/ScriptEditorMenu.java]: Its awkward to use other stuff

Comment (Reasonlesss) [reply common/src/main/java/io/github/codeutilities/menu/ScriptEditorMenu.java]: Legacy is a big no, I will stand by this.

Comment (TechStreetDev) [reply common/src/main/java/io/github/codeutilities/menu/ScriptEditorMenu.java]: What other options do we have?

Comment (Reasonlesss) [reply common/src/main/java/io/github/codeutilities/menu/ScriptEditorMenu.java]: Components, actually rendering colored text the intended way.

Comment (TechStreetDev) [reply common/src/main/java/io/github/codeutilities/menu/ScriptEditorMenu.java]: Also there not legacy in architectury @Reasonlesss it readds them and adds §x codes

**Review by Reasonlesss — APPROVED** [review]

**Review by TechStreetDev — APPROVED** [review]

**Pull request merged**